src/doc_table.c

```diff
@@ -134,39 +180,38 @@ t_docId DocTable_Put(DocTable *t, RSDocumentKey key, double score, u_char flags,
 
   sds keyPtr = sdsnewlen(key.str, key.len);
 
-  t->docs[docId] = (RSDocumentMetadata){.keyPtr = keyPtr,
-                                        .score = score,
-                                        .flags = flags,
-                                        .payload = dpl,
-                                        .maxFreq = 1,
-                                        .sortVector = NULL};
+  RSDocumentMetadata dmd = (RSDocumentMetadata){.keyPtr = keyPtr,
+                                                .score = score,
+                                                .flags = flags,
+                                                .payload = dpl,
+                                                .maxFreq = 1,
+                                                .id = docId,
+                                                .sortVector = NULL};
+  DocTable_Set(t, docId, dmd);
   ++t->size;
   t->memsize += sizeof(RSDocumentMetadata) + sdsAllocSize(keyPtr);
   DocIdMap_Put(&t->dim, key, docId);
   return docId;
 }
 
 RSPayload *DocTable_GetPayload(DocTable *t, t_docId docId) {
-  if (docId == 0 || docId > t->maxDocId) {
-    return NULL;
-  }
-  return t->docs[docId].payload;
+  RSDocumentMetadata *dmd = DocTable_Get(t, docId);
+  return dmd ? dmd->payload : NULL;
```

Comment:
Rather than returning NULL and doing all these checks, have a blank DMD (static memory?) that you can just return as a default which contains all 0s. Makes the code change smaller and the runtime faster. Besides I'm sure that all the calling code already checks for != 0

Reply:
lots of parts in the code already check for that the return value is NULL and if so it assume the doc does not exists, not sure that we should change it.